tests/ztest/busy_sim/boards/nrf52840dk_nrf52840_pin.overlay

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&timer0 {
+	status = "okay";
+	interrupts = <8 0>;
+	busy_sim: vnd-busy-sim {
+		compatible = "vnd,busy-sim";
+		active-gpios = <&gpio0 27 GPIO_ACTIVE_HIGH>;
+	};
+};
```

Comment:
Can you please provide a justification for:

1. why you're using a parent/child relationship here instead of pointing at the timer instance from the vnd,busy-sim node using a phandle?
2. why only one device is supported via the busy_sim node label?

If you can support multiple busy simulator devices, I think this should use DT_INST_FOREACH_STATUS_OKAY to initialize them instead, with DT_DRV_COMPAT set to vnd_busy_sim.

If it only makes sense to support one vnd,busy-sim in the system, then why not just use DEVICE_DT_GET_ANY() and add a `BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(vnd_busy_sim) == 1)`?

Reply:
> why you're using a parent/child relationship here instead of pointing at the timer instance from the vnd,busy-sim node using a phandle?

I am not that fluent with device tree options. This one seemed ok to me as busy simulator is built on top of counter instance but i'm not attached to that so if you think it's better to use phandle i can change it. Could you provide justification why it's the way to go?

> why only one device is supported via the busy_sim node label?

Since it's for testing purposes only i don't see a need for multiple simulators? Sorry but don't know where should i use DEVICE_DT_GET_ANY(). Can you be more specific?

Reply:
> Could you provide justification why it's the way to go?

In my experience, parent/child is for controller / peripheral type relationships (compare with SPI, I2C, and flash).

Phandles seem to be used more when you want to say  "use peripheral X for this": compare with properties like `foo-gpios`, `dmas`, `io-channels`, etc.

> Sorry but don't know where should i use DEVICE_DT_GET_ANY(). Can you be more specific?

Sure; what I mean is that you don't need to force the user to make a node label, even if only one device is supported in the system.

You can do something like this instead (after thinking about it, I think DEVICE_DET_GET_ONE is better than DEVICE_DT_GET_ANY):

```c
BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(vnd_busy_sim) == 1, "add exactly one vnd,busy-sim node to the devicetree");
const struct device *my_busy_sim = DEVICE_DT_GET_ONE(vnd_busy_sim);
if (!device_is_ready(my_busy_sim)) {
        /* ... handle error here ... */
} else {
        /* device is good to go */
}
```

Then your node doesn't need to be labeled `busy_sim` in the DTS. You can just make a status `okay` node with the right compatible and everything just works.

Reply:
I've converted busy_sim to the device and used phandle for counter. Implemented it as separate commit, will squash if it's ok.